src/factory.rs

```diff
@@ -397,6 +400,35 @@ impl Factory {
             target,
         }
     }
+
+    /// Create a basic mesh pipeline using a custom shader.
+    pub fn basic_pipeline(&mut self,
+        shader_path: &str,
+        primitive: gfx::Primitive,
+        rasterizer: gfx::state::Rasterizer,
+        depth_state: gfx::state::Depth,
+    ) -> Result<BasicPipelineState, ()> {
+        let code_vs = get_shader(&self.root_shader_path, shader_path, "vs");
+        let code_ps = get_shader(&self.root_shader_path, shader_path, "ps");
+
+        let program = match self.backend.link_program(code_vs.as_bytes(), code_ps.as_bytes()) {
```

Comment:
This is not the case for `unwrap_or_else`, because the error branch actually returns an error.